plugins/language_kotlin.lua

```diff
@@ -0,0 +1,119 @@
+-- mod-version:3
+
+-- https://kotlinlang.org/docs/keyword-reference.html
+
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "Kotlin",
+  files = { "%.kt$" },
+  comment = "//",
+  patterns = {
+    { pattern = "//.*",                        type = "comment"  }, -- Comment, single-line
+    { pattern = { "/%*", "%*/" },              type = "comment"  }, -- Comment, multi-line
+    { pattern = { '"', '"', '\\' },            type = "string"   }, -- String, quotation marks
+    { pattern = { "'", "'", '\\' },            type = "string"   }, -- String, apices
+    { pattern = "'\\x%x?%x?%x?%x'",            type = "string"   }, -- Character hexadecimal escape sequence
+    { pattern = "'\\u%x%x%x%x'",               type = "string"   }, -- Character unicode escape sequence
+    { pattern = "'\\?.'",                      type = "string"   }, -- Character literal
+    { pattern = "-?0x%x+",                     type = "number"   }, -- ?
+    { pattern = "-?%d+[%deE]*f?",              type = "number"   }, -- ?
+    { pattern = "-?%.?%d+f?",                  type = "number"   }, -- ?
+    { regex = "\\-\\>(?=\\s)",                 type = "operator" },  -- Lambda
+    { regex = "\\.{2}\\<?\\s?(?=[a-z0-9])",    type = "operator" }, -- Range operators
```

Comment:
I suggest using raw strings for this:
```suggestion
    { regex = [[\-\>(?=\s)]],                  type = "operator" },  -- Lambda
    { regex = [[\.{2}\<?\s?(?=[a-z0-9])]],     type = "operator" }, -- Range operators
```
as it'll make escaping more readable.